plugin-server/src/ingestion/ingestion-consumer.test.ts

```diff
@@ -0,0 +1,265 @@
+import { DateTime } from 'luxon'
+import { Message } from 'node-rdkafka'
+
+import { UUIDT } from '~/src/utils/utils'
+import {
+    getProducedKafkaMessages,
+    getProducedKakfaMessagesForTopic,
+    mockProducer,
+} from '~/tests/helpers/mocks/producer.mock'
+import { forSnapshot } from '~/tests/helpers/snapshots'
+import { createTeam, getFirstTeam, resetTestDatabase } from '~/tests/helpers/sql'
+
+import { Hub, PipelineEvent, Team } from '../../src/types'
+import { closeHub, createHub } from '../../src/utils/db/hub'
+import { status } from '../utils/status'
+import { IngestionConsumer } from './ingestion-consumer'
+
+const mockConsumer = {
+    on: jest.fn(),
+    commitSync: jest.fn(),
+    commit: jest.fn(),
+    queryWatermarkOffsets: jest.fn(),
+    committed: jest.fn(),
+    assignments: jest.fn(),
+    isConnected: jest.fn(() => true),
+    getMetadata: jest.fn(),
+}
+
+jest.mock('../../src/kafka/batch-consumer', () => {
+    return {
+        startBatchConsumer: jest.fn(() =>
+            Promise.resolve({
+                join: () => ({
+                    finally: jest.fn(),
+                }),
+                stop: jest.fn(),
+                consumer: mockConsumer,
+            })
+        ),
+    }
+})
+
+jest.setTimeout(1000)
+
+let offsetIncrementer = 0
+
+const createKafkaMessages: (events: PipelineEvent[]) => Message[] = (events) => {
+    return events.map((event) => {
+        // TRICKY: This is the slightly different format that capture sends
+        const captureEvent = {
+            uuid: event.uuid,
+            distinct_id: event.distinct_id,
+            ip: event.ip,
+            now: event.now,
+            token: event.token,
+            data: JSON.stringify(event),
+        }
+        return {
+            value: Buffer.from(JSON.stringify(captureEvent)),
+            size: 1,
+            topic: 'test',
+            offset: offsetIncrementer++,
+            partition: 1,
+        }
+    })
+}
+
+describe('IngestionConsumer', () => {
+    let ingester: IngestionConsumer
+    let hub: Hub
+    let team: Team
+    let team2: Team
+
+    const createEvent = (event?: Partial<PipelineEvent>): PipelineEvent => ({
+        distinct_id: 'user-1',
+        uuid: new UUIDT().toString(),
+        token: team.api_token,
+        ip: '127.0.0.1',
+        site_url: 'us.posthog.com',
+        now: DateTime.now().toISO(),
+        event: '$pageview',
+        properties: {
+            $current_url: 'http://localhost:8000',
+        },
+        ...event,
+    })
+
+    beforeEach(async () => {
+        const now = new Date(2025, 1, 1).getTime()
+        jest.spyOn(Date, 'now').mockReturnValue(now)
```

Comment:
issue: This looks like it's gonna break when you run it in a different timezone – in the snapshot it's already a different date. I assume we don't want to mock the timers using jest because it's an integration test and it might be running timers somewhere in the middle of the stack. Could we do the same as we do with UUIDs or completely mask out the timestamps in the snapshots?

Reply:
Yeah I tried a few things and this seemed to be the best option. I'll circle back and see if I can make it nicer

Reply:
Left it as is as it works. I don't want to mock out the snapshots like the UUID as it is important that timestamps are correctly verifiable at a glance